lib/compress/zstd_compress.c

```diff
@@ -2050,6 +2053,10 @@ MEM_STATIC size_t ZSTD_compressSequences_internal(seqStore_t* seqStorePtr,
                     workspace, HUF_WORKSPACE_SIZE);
             if (ZSTD_isError(countSize)) return countSize;
             op += countSize;
+            if (Offtype == set_compressed)
+                lastCountSize = countSize;
+            else if (countSize > 0)
```

Comment:
I would have expected : 
`else lastCountSize += countSize;`

Reply:
That would work too, and it makes the check a bit simpler.